server/server/src/billing/stripe_client.rs

```diff
@@ -0,0 +1,437 @@
+use crate::billing::stripe::{
+    BasicStripeResponse, SetupPaymentIntentStatus, StripeError, StripeErrorType, StripeInvoice,
+    StripeKnownErrorCode, StripeKnownErrorDeclineCode, StripeMaybeContainer,
+    StripePaymentMethodResponse, StripeResult, StripeSetupIntentResponse,
+    StripeSubscriptionResponse, SubscriptionStatus,
+};
+use crate::billing::stripe_client::StripeClientError::{CardDeclined, InvalidCreditCard, Other};
+use crate::ServerState;
+use lockbook_models::api::{CardDeclinedType, InvalidCreditCardType};
+use log::error;
+use reqwest::Method;
+use serde::de::DeserializeOwned;
+use std::collections::HashMap;
+use std::fmt::Debug;
+
+static STRIPE_ENDPOINT: &str = "https://api.stripe.com/v1";
+static PAYMENT_METHODS_ENDPOINT: &str = "/payment_methods";
+static DETACH_ENDPOINT: &str = "/detach";
+static ATTACH_ENDPOINT: &str = "/attach";
+static CUSTOMER_ENDPOINT: &str = "/customers";
+static SUBSCRIPTIONS_ENDPOINT: &str = "/subscriptions";
+static SETUP_INTENTS_ENDPOINT: &str = "/setup_intents";
+static INVOICES_ENDPOINT: &str = "/invoices";
+
+#[derive(Debug)]
+pub enum StripeClientError {
+    CardDeclined(CardDeclinedType),
+    InvalidCreditCard(InvalidCreditCardType),
+    Other(String),
+}
+
+pub async fn create_customer(server_state: &ServerState) -> Result<String, StripeClientError> {
+    match send_stripe_request::<BasicStripeResponse>(
```

Comment:
![image](https://user-images.githubusercontent.com/821972/150005418-0df8c091-7e0a-442e-a875-b97ed7a31052.png)
These are currently showing up like this, maybe we want to save a UUID as a name and save that in our DB too. You can leave this as a last thing for us to discuss.